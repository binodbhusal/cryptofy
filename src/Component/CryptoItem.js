/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

import styles from './CryptoItem.module.css';

const CryptoItem = ({ cryptoData }) => {
  const navigate = useNavigate();
  const {
    name, rank, volumeUsd, exchangeUrl, exchangeId,
  } = cryptoData;
  let faviconUrl = '';
  try {
    const domain = new URL(exchangeUrl).hostname;
    faviconUrl = `https://www.google.com/s2/favicons?domain=${domain}`;
  } catch (error) {
    faviconUrl = 'error-favicon-url';
  }
  const formatdVolumeUsd = Number(volumeUsd).toFixed(2);
  const handleExpandClick = () => {
    navigate(`/crypto-details/${exchangeId}`);
  };

  return (
    <div className={styles.card}>
      {' '}
      <img src={faviconUrl} alt={`${name}`} style={{ width: '20%', padding: '0 8px', marginTop: '10%' }} />
      <button type="button" onClick={handleExpandClick} className={styles.expandBtn}>
        <img width="25" height="25" src="https://img.icons8.com/ios/25/ffffff/circled-right-2.png" alt="circled-right-2" />
      </button>
      <div className={styles.content}>
        <h5>{name}</h5>
        <p>
          Rank: #
          {' '}
          {rank}
        </p>
        <p>
          $
          {' '}
          {formatdVolumeUsd}
        </p>
        {' '}

      </div>
    </div>
  );
};

CryptoItem.propTypes = {
  cryptoData: PropTypes.shape({
    name: PropTypes.string.isRequired,
    rank: PropTypes.string.isRequired,
    exchangeUrl: PropTypes.string.isRequired,
    volumeUsd: PropTypes.string.isRequired,
    exchangeId: PropTypes.string.isRequired,
  }).isRequired,
};

export default CryptoItem;
