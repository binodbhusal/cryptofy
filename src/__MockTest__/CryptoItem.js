import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

const CryptoItem = ({ cryptoData }) => {
  const navigate = useNavigate();
  const {
    name, rank, volumeUsd, exchangeUrl, exchangeId,
  } = cryptoData;
  parseInt(cryptoData.rank, 10);

  let faviconUrl = '';
  try {
    const domain = new URL(exchangeUrl).hostname;
    faviconUrl = `https://www.google.com/s2/favicons?domain=${domain}`;
  } catch (error) {
    faviconUrl = 'error-favicon-url';
  }
  const formatdVolumeUsd = Number(volumeUsd).toFixed(2);
  const handleExpandClick = () => {
    navigate(`/crypto-details/${exchangeId}`);
  };
  return (
    <li>
      <img src={faviconUrl} alt={`${name} favicon`} />
      <p>{name}</p>
      <p>{rank}</p>
      <p>
        $
        {' '}
        {formatdVolumeUsd}
      </p>
      <button type="button" onClick={handleExpandClick}>
        Expand
      </button>
    </li>
  );
};

CryptoItem.propTypes = {
  cryptoData: PropTypes.shape({
    name: PropTypes.string,
    rank: PropTypes.number,
    exchangeUrl: PropTypes.string,
    volumeUsd: PropTypes.string,
    exchangeId: PropTypes.string,
    expanded: PropTypes.string,
  }).isRequired,
};

export default CryptoItem;
