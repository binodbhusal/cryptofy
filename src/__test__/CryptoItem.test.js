import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CryptoItem from '../__MockTest__/CryptoItem';
import store from '../__MockTest__/mockstore';

describe('Crypto Item component', () => {
  test('renders crypto item and handle expand details button', () => {
    const cryptoData = {
      name: 'Binance',
    };
    const { getByText } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CryptoItem cryptoData={cryptoData} />
        </Provider>
      </MemoryRouter>,
    );
    const exchangeName = getByText('Binance');
    expect(exchangeName).toBeInTheDocument();
    expect(exchangeName).toMatchSnapshot();
  });
});

// describe('RocketItem Component', () => {
//   const mockStore = configureStore();
//   const initialState = {
//   };
//   const store = mockStore(initialState);

//   test('renders rocket item and handles reserve/cancel reserve', () => {
//     const rocketData = {
//       id: '1',
//       rocket_name: 'Falcon 9',
//       description: 'Sample rocket description',
//       flickr_images: ['image1.jpg'],
//       reserved: true,
//     };

//     const { getByText } = render(
//       <MemoryRouter>
//         <Provider store={store}>
//           <RocketItem rocketData={rocketData} />
//         </Provider>
//       </MemoryRouter>,
//     );

//     const rocketName = getByText('Falcon 9');
//     expect(rocketName).toBeInTheDocument();

//     const rocketDescription = getByText('Sample rocket description');
//     expect(rocketDescription).toBeInTheDocument();
// eslint-disable-next-line max-len
//     const cancelReserveButton = getByText((content, element) => element.tagName.toLowerCase() === 'button' && content === 'Cancel Reserve');
//     expect(cancelReserveButton).toBeInTheDocument();
//     fireEvent.click(cancelReserveButton);
//   });
// });
