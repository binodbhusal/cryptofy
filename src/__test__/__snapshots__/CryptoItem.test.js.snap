// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crypto Item component renders crypto item and handle expand details button 1`] = `
<p>
  Binance
</p>
`;
